example-code/snippets/src/backend/ExampleCode.test.ts

```diff
@@ -201,6 +201,60 @@ describe("Example Code", () => {
     // __PUBLISH_EXTRACT_END__
   });
 
+  it("CodeService", async () => {
+
+    if (false) { // this will compile but it will not run, because the root elementhas no federationGuid -- waiting for a fix
+
+      // __PUBLISH_EXTRACT_START__ CodeService.reserveInternalCodeForNewElement
+      const code = Subject.createCode(iModel, IModel.rootSubjectId, "test"); // example code
+
+      const proposedCode = CodeService.makeProposedCode({ iModel, code, props: { guid: Guid.createValue() } });
+      await iModel.codeService?.internalCodes?.writeLocker.reserveCode(proposedCode);
```

Comment:
you should either add a try/catch here or note that this function will throw and not insert a new subject if someone else has already added a subject named "test" (probably a better name would help).